bcda/auth/middleware.go

```diff
@@ -58,48 +59,71 @@ func ParseToken(next http.Handler) http.Handler {
 
 		tokenString := authSubmatches[1]
 
-		token, err := GetProvider().VerifyToken(tokenString)
+		token, ad, err := AuthorizeAccess(tokenString)
 		if err != nil {
-			log.Auth.Errorf("Unable to verify token; %s", err)
 			handleTokenVerificationError(w, rw, err)
 			return
 		}
 
-		var ad AuthData
-		if claims, ok := token.Claims.(*CommonClaims); ok && token.Valid {
-			switch claims.Issuer {
-			case "ssas":
-				ad, err = GetProvider().getAuthDataFromClaims(claims)
-				if err != nil {
-					handleSsasAuthDataError(w, rw, err)
-					return
-				}
-			default:
-				log.Auth.Errorf("Unsupported claims issuer %s", claims.Issuer)
-				rw.Exception(w, http.StatusNotFound, responseutils.TokenErr, "")
-				return
-			}
-		}
 		ctx := context.WithValue(r.Context(), TokenContextKey, token)
 		ctx = context.WithValue(ctx, AuthDataContextKey, ad)
 		next.ServeHTTP(w, r.WithContext(ctx))
 	})
 }
 
-func handleSsasAuthDataError(w http.ResponseWriter, rw fhirResponseWriter, err error) {
-	log.Auth.Error(err)
-	if _, ok := err.(*customErrors.EntityNotFoundError); ok {
-		rw.Exception(w, http.StatusForbidden, responseutils.UnauthorizedErr, responseutils.UnknownEntityErr)
-	} else {
-		rw.Exception(w, http.StatusUnauthorized, responseutils.TokenErr, "")
+// AuthorizeAccess asserts that a base64 encoded token string is valid for accessing the BCDA API.
+func AuthorizeAccess(tokenString string) (*jwt.Token, AuthData, error) {
```

Comment:
AuthorizeAccess was basically doing the same validation as ParseToken with some additional logging, so I merged the two to create this helper function that is now used by ParseToken.